src/Tests/Microsoft.NET.Build.Tests/AppHostTests.cs

```diff
@@ -76,35 +96,85 @@ public void It_builds_a_runnable_apphost_if_opt_in_on_mac()
             var buildCommand = new BuildCommand(testAsset);
             buildCommand
                 .Execute(new string[] {
-                    "/restore", "/p:UseAppHost=true",
+                    "/p:_EnableMacOSCodeSign=false",
                 })
                 .Should()
                 .Pass();
 
             var outputDirectory = buildCommand.GetOutputDirectory(targetFramework);
             var hostExecutable = $"HelloWorld{Constants.ExeSuffix}";
+            var appHostFullPath = Path.Combine(outputDirectory.FullName, hostExecutable);
 
-            outputDirectory.Should().OnlyHaveFiles(new[] {
-                hostExecutable,
-                "HelloWorld.dll",
-                "HelloWorld.pdb",
-                "HelloWorld.deps.json",
-                "HelloWorld.runtimeconfig.dev.json",
-                "HelloWorld.runtimeconfig.json",
-            });
+            // Check that the apphost was not signed
+            var psi = new ProcessStartInfo()
+            {
+                Arguments = $"-d {appHostFullPath}",
+                FileName = @"/usr/bin/codesign",
+                RedirectStandardError = true
+            };
+
+            using (var codesign = Process.Start(psi))
+            {
+                codesign.Start();
+                codesign.StandardError.ReadToEnd()
```

Comment:
I'd suggest using `RunExeCommand` here, which handles running the process and capturing the output.